lib/upgrade/migrator/clients/v1/etcdv2/etcdv2.go

```diff
@@ -34,6 +35,8 @@ var (
 	etcdGetOpts          = &etcd.GetOptions{Quorum: true}
 	etcdListOpts         = &etcd.GetOptions{Quorum: true, Recursive: true, Sort: true}
 	etcdListChildrenOpts = &etcd.GetOptions{Quorum: true, Recursive: false, Sort: true}
+	etcdCreateOpts       = &etcd.SetOptions{PrevExist: etcd.PrevNoExist}
+	etcdDeleteEmptyOpts  = &etcd.DeleteOptions{Recursive: false, Dir: true}
```

Comment:
What happens if we were to delete a node without deleting its children? Do the children get orphaned or is there some other way to access the children after its parent node has been deleted?

Reply:
TBH I didn't really consider the options (other than that the Create will error if it already exists). I copied the Delete from https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/etcd/etcd.go#L177, and the etcdDeleteEmptyOpts is needed for that one.  I think the Delete could be cut down a lot but I went with just copying something, that I assume worked, instead of trying to write it myself.

Reply:
Looks good, just looked it up, it deletes only if the directory at the path is empty.